### PR TITLE
Fix issues with using API keys

### LINKED ISSUE #8: .grist-api-key shouldn't be at ~/
mainly because you can't test the npm package on replit, codesandbox, glitch, ...
and also because not all js are run under a user with a ~ folder.

image

/home/runner is on replit server, pretty sure nobody can reach it :smile: 

Also, since "everyone" user has read access, it should work without it.
It works in postman without any authentication.

### PR DESCRIPTION
**Pull request by dsagal**
Resolves #8.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ npm install grist-api
 const {GristDocAPI} = require('grist-api');
 
 // Put here the URL of your document.
-const DOC_URL = "https://subdomain.getgrist.com/doc/12345678-2761-4ef2-bc28-310e634754fb";
+const DOC_URL = "https://<subdomain>.getgrist.com/doc/12345678-2761-4ef2-bc28-310e634754fb";
```
Comment (paulfitz) [inline README.md:24]: Most document urls don't look like this any more, how about freshening it to something like /123456789abc/My-Document?

File: README.md [hunk 2]
```diff
@@ -52,6 +52,8 @@ To use the API, you need to get your API key in Grist from Profile Settings. Run
 with `GRIST_API_KEY=<key>` in the shell environment. The key may also be stored to
 `~/.grist-api-key` file.
 
+To access public documents, you may provide an empty string for the API key.
```
Comment (paulfitz) [inline README.md:55]: Is it ok to just not set it?

File: lib/grist-api.ts [hunk 3]
```diff
@@ -7,6 +7,10 @@ import chunk = require('lodash/chunk');
 import mapValues = require('lodash/mapValues');
 import pick = require('lodash/pick');
 
+// Require type only, since the actual require may not be needed or available,
+// depending on how and where grist-api is used.
+import type * as FsExtra from 'fs-extra';
```
Comment (paulfitz) [inline lib/grist-api.ts:12]: Sneaky!

File: lib/grist-api.ts [hunk 4]
```diff
@@ -30,17 +34,17 @@ export async function getAPIKey(): Promise<string> {
   }
 
   // Otherwise, assume we are in node environment.
-  if (process.env.GRIST_API_KEY) {
+  if (process.env.GRIST_API_KEY !== undefined) {
```
Comment (paulfitz) [inline lib/grist-api.ts:37]: Oh I see, so you set it to blank in the environment, so that the code doesn't reach the file system check? Makes some sense.

**Review by paulfitz — APPROVED** [review]
It works, and is better. Reading a key from the file system is making things a bit slippery though, perhaps there could be a distinction between the library and a command line tool (or more specifically an entrypoint designed for use in clis)

File: README.md (hunk 2 above)
Comment (dsagal) [reply README.md:55]: Yes. Rephrased this, and the above paragraph, to clarify.

**Comment by paulfitz** [comment]
Thanks, looks good.

**Pull request merged**